src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/client/CommunityApiClient.kt

```diff
@@ -27,14 +25,18 @@ import java.time.Duration
 class CommunityApiClient(
   @Qualifier("communityApiWebClient") private val webClient: WebClient,
   objectMapper: ObjectMapper,
-  redisTemplate: RedisTemplate<String, String>,
-  @Value("\${preemptive-cache-key-prefix}") preemptiveCacheKeyPrefix: String,
-) : BaseHMPPSClient(webClient, objectMapper, redisTemplate, preemptiveCacheKeyPrefix) {
-  private val offenderDetailCacheConfig = PreemptiveCacheConfig(
+  webClientCache: WebClientCache,
+) : BaseHMPPSClient(webClient, objectMapper, webClientCache) {
+  private val offenderDetailCacheConfig = WebClientCache.PreemptiveCacheConfig(
```

Comment:
If we wanted to add an environment variable to feature flag caching so we could quickly turn it off, would the best way be to set this variable in each API client to null?